apps/src/music/views/TimelineSimple2Events.tsx

```diff
@@ -0,0 +1,152 @@
+import React, {useCallback, useMemo} from 'react';
+import TimelineElement from './TimelineElement';
+import {useMusicSelector} from './types';
+import {FunctionEvents} from '../player/interfaces/FunctionEvents';
+
+/**
+ * Compute the bounds for the given function, given the list of unique sounds and all functions.
+ */
+const getFunctionBounds = (
+  orderedFunction: FunctionEvents,
+  uniqueSounds: string[],
+  orderedFunctions: FunctionEvents[]
+) => {
+  let left = Number.MAX_SAFE_INTEGER,
+    top = Number.MAX_SAFE_INTEGER,
+    right = 0,
+    bottom = 0;
+
+  orderedFunction.playbackEvents.forEach(playbackEvent => {
```

Comment:
Out of curiosity, do you have a preference for `forEach` over `for...of`?

Reply:
Oh no preference, I think this function was copied as-is - can switch to `for...of` if you think that's clearer!

Reply:
I used to have a slight perference for `for...of`, partly because I think it used to be easier to step into with a browser debugger, but these days either seems fine.